inc/class-sitemaps-index.php

```diff
@@ -60,4 +61,18 @@ public function output_sitemap() {
 			exit;
 		}
 	}
+
+	/**
+	 * Adds the sitemap index to robots.txt.
+	 *
+	 * @param string $output robots.txt output.
+	 * @param bool   $public Whether the site is public or not.
+	 * @return string robots.txt output.
+	 */
+	public function add_robots( $output, $public ) {
+		if ( $public ) {
+			$output .= 'Sitemap: ' . home_url( '/sitemap.xml' ) . "\n";
```

Comment:
The argument to home_url() is dependant on the rewrite rule added by the index class, so ideally we want to transform one to the other so that if one is changed the robots.txt does not break. Also home_url uses get_home_url() which value can be replaced using the home_url filter so I would suggest to sanitize the value using esc_url().